### PR TITLE
Admin Sidebar Menu Persistence

### PR DESCRIPTION
**Pull request by davidangel**
Also, disable dimming and autoclosing of the sidebar.

| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

**Comment by davidangel** [comment]
I think this automatically closed when I was trying to squash the commits.

**Comment by sweoggy** [comment]
Why use a JS library for cookie support when you can just use localStorage? No need to bloat Sylius :)

Great work apart from that! :+1:

File: src/Sylius/Bundle/UiBundle/Resources/private/js/app.js [hunk 1]
```diff
@@ -9,10 +9,21 @@
...
-        .sidebar('attach events', '#sidebar-toggle', 'show')
-    ;
+    var $primarySidebar = $('#sidebar').first();
+    $primarySidebar.sidebar('setting', { dimPage: false, closable: false });
+    if(Cookies.get('sidebar_visible') === 'true') {
+        $primarySidebar.addClass('visible');
+    }
+    $('#sidebar-toggle').click(function (e) {
+        if(Cookies.get('sidebar_visible') !== 'true') {
+            Cookies.set('sidebar_visible', 'true', { path: '/' });
```
Comment (sweoggy) [inline src/Sylius/Bundle/UiBundle/Resources/private/js/app.js]: localStorage.setItem('sidebar_visible', 'true');

File: src/Sylius/Bundle/UiBundle/Resources/private/js/app.js [hunk 2]
```diff
@@ -9,10 +9,21 @@
...
+            $primarySidebar.sidebar('show');
+        } else {
+            Cookies.remove('sidebar_visible');
+            Cookies.set('sidebar_visible', 'false', { path: '/' });
```
Comment (sweoggy) [inline src/Sylius/Bundle/UiBundle/Resources/private/js/app.js]: localStorage.setItem('sidebar_visible', 'false');

File: src/Sylius/Bundle/UiBundle/Resources/private/js/app.js [hunk 3]
```diff
@@ -9,10 +9,21 @@
...
-        .first()
-        .sidebar('attach events', '#sidebar-toggle', 'show')
-    ;
+    var $primarySidebar = $('#sidebar').first();
+    $primarySidebar.sidebar('setting', { dimPage: false, closable: false });
+    if(Cookies.get('sidebar_visible') === 'true') {
+        $primarySidebar.addClass('visible');
+    }
+    $('#sidebar-toggle').click(function (e) {
+        if(Cookies.get('sidebar_visible') !== 'true') {
```
Comment (sweoggy) [inline src/Sylius/Bundle/UiBundle/Resources/private/js/app.js]: localStorage.getItem('sidebar_visible') !== 'true'

**Comment by davidangel** [comment]
@sweoggy – Originally I was thinking of accessing cookie values from twig templates, because I was concerned about adding the 'visible' class to the sidebar via JS. But that's how I've implemented that bit after all, so you're correct. 😜  I'll have at it using localStorage.

File: src/Sylius/Bundle/UiBundle/Resources/private/js/app.js [hunk 4]
```diff
@@ -9,10 +9,21 @@
 
 (function($) {
   $(document).ready(function() {
-    $('#sidebar')
-        .first()
-        .sidebar('attach events', '#sidebar-toggle', 'show')
-    ;
+    var $primarySidebar = $('#sidebar').first();
+    $primarySidebar.sidebar('setting', { dimPage: false, closable: false });
+    if(Cookies.get('sidebar_visible') === 'true') {
```
Comment (sweoggy) [inline src/Sylius/Bundle/UiBundle/Resources/private/js/app.js]: localStorage.getItem('sidebar_visible') === 'true'

**Comment by pjedrzejewski** [comment]
Nice work David, thank you so much! :)

**Pull request merged**